### PR TITLE
HOTFIX: Fix bug in readToLogEnd in KafkaBasedLog.

### PR DESCRIPTION
**Pull request by kkonstantine**

### CODE REVIEW

**Comment by kkonstantine** [comment]
Now that the code changed to collect the end offsets beforehand for all partitions, we need to first check whether there are any unread messages before we issue a blocking call to poll.

File: connect/runtime/src/test/java/org/apache/kafka/connect/util/KafkaBasedLogTest.java [hunk 1]
```diff
@@ -219,6 +220,43 @@ public void run() {
...
+                // time outs (for instance via ConnectRestException)
+                throw new WakeupException();
+            }
+        });
+
+        store.start();
+
+        assertEquals(CONSUMER_ASSIGNMENT, consumer.assignment());
+        assertEquals(7L, consumer.position(TP0));
+        assertEquals(7L, consumer.position(TP0));
```
Comment (ijuma) [inline connect/runtime/src/test/java/org/apache/kafka/connect/util/KafkaBasedLogTest.java]: Should this be TP1?

**Review by ijuma — COMMENTED** [review]
Left one comment. Seems fine otherwise, but I'll leave it to @ewencp to review as well since I don't know connect's code well.

File: connect/runtime/src/test/java/org/apache/kafka/connect/util/KafkaBasedLogTest.java (hunk 1 above)
Comment (kkonstantine) [reply connect/runtime/src/test/java/org/apache/kafka/connect/util/KafkaBasedLogTest.java]: Yes!

**Comment by kkonstantine** [comment]
Thanks, I appreciate the review @ijuma

**Comment by ewencp** [comment]
LGTM, thanks!

**Pull request closed without merging**